Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -337,21 +373,24 @@ std::pair<String, MarkupAccumulator::IsCreatedByURLReplacement> MarkupAccumulato
 
 RefPtr<Element> MarkupAccumulator::replacementElement(const Node& node)
 {
-    if (!m_shouldIncludeShadowDOM)
+    if (!shouldIncludeShadowRoots())
         return nullptr;
 
     RefPtr shadowRoot = dynamicDowncast<ShadowRoot>(node);
-    if (!shadowRoot)
-        return nullptr;
-
-    if (shadowRoot->mode() == ShadowRootMode::UserAgent)
+    if (!shadowRoot || !includeShadowRoot(*shadowRoot))
         return nullptr;
 
     auto element = HTMLTemplateElement::create(HTMLNames::templateTag, node.protectedDocument());
```

Comment:
This is an awfully inefficient way of serializing shadow root.
We should probably cleanup in a follow up to avoid mallocing a phantom element like this.